7xbot.py

```
@@ -501,6 +532,74 @@ async def ggb(ctx, state: Optional[str] = None):
+
+    async def ask(prompt: str, parser):
+        await ctx.send(prompt)
+        try:
+            msg = await bot.wait_for(
+                "message",
+                check=lambda m: m.author == ctx.author and m.channel == ctx.channel,
+                timeout=60,
+            )
+        except asyncio.TimeoutError:
+            await ctx.send("Timed out waiting for a response. Setup cancelled.")
+            raise SetupCancelled
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)